nim-suggest.el

```diff
@@ -55,6 +55,10 @@
     :prefix) ; matching state. See also prefixmatches.nim
   "Keywords for SexpNode type on nimsuggest.nim.")
 
+(defvar nimsuggest-accept-process-timeout 150)
+(defvar nimsuggest-accept-process-timeout-count 100)
```

Comment:
Can you add a documenation string for these variables? For me it's not clear what unit the timeout is, and what the count is actually counting.

Reply:
will do. It's just timeout values how long you can wait if nimsuggest doesn't answer for its port for first connection.

Reply:
I meant until emacs side gets port number for nimsuggest server. In current implementation, if the timeout happen, it just block the query to nimsuggest for the buffer, so annoying frozen should not occur again in the buffer. I hope.

Reply:
I think you shouldn't block emacs for 150ms. Just assume that the start of nimsuggest needs to be async. On a 60Hz monitor 150ms is 9 frames of emacs hanging there doing nothing.

Reply:
I think you are misunderstanding. the `deferred-wait` doesn't stop Emacs IIUC